tools/JSTooling.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * @license
+ * Copyright 2025 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+const pmake    = require('./pmake.js');
```

Comment:
as discussed in chat, we should find a way to inject these stuff into the tasks directly instead of needing to use require().

That way these pom files can look the same as the pom files in the source tree and still be loaded on in other environmetns that don't provide require() (such as web)


I think the existing build.js tried to achieve this by setting "this" to an object that contains all the buildlib functions when running the pom tasks so that the tasks can do this `this.execSync` and whatnot